### PR TITLE
malta: disable unnecessary built-in CDROM support

### PR DESCRIPTION
**Pull request by guidosarducci**
**Review/Distribution List**:  @dangowrt, @yousong  (based on similar fixes and commits in the repo)
**Compile tested**:  mips32-be-malta and mips64-le-malta, master branch
**Run tested**:  mips32-be-malta and mips64-le-malta, master branch

**Description**:
The CDROM is not needed for booting and can be included by selecting the
loadable module as a package instead.

This also avoids triggering a memory allocation failure during probing of
the CDROM due to lack of low 16MB DMA memory, as decribed in FS#3278:
https://bugs.openwrt.org/index.php?do=details&task_id=3278

Appreciate your review and feedback!

### CODE REVIEW

**Comment by yousong** [comment]
Likely a qemu issue.  I disabled cdrom emulation and it boots fine without the "out of memory" issue.

**Comment by yousong** [comment]
LGTM

Better just drop out kernel cdrom support here.  Real malta board also does not have that and we do not use it.

File: target/linux/malta/config-5.4
Comment (yousong) [inline target/linux/malta/config-5.4:37]: With `kernel_menuconfig`, these lines should be dropped.  This will do when other people does this in future config refreshes.

fyi,

```
--- a/target/linux/malta/config-5.4
+++ b/target/linux/malta/config-5.4
@@ -33,15 +33,12 @@ CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=16
 CONFIG_BLK_DEV_RAM_SIZE=4096
 CONFIG_BLK_DEV_SD=y
-CONFIG_BLK_DEV_SR=y
-# CONFIG_BLK_DEV_SR_VENDOR is not set
 CONFIG_BLK_MQ_PCI=y
 CONFIG_BLK_SCSI_REQUEST=y
 CONFIG_BOARD_SCACHE=y
 CONFIG_BOOT_ELF32=y
 CONFIG_BOUNCE=y
 CONFIG_BUILTIN_DTB=y
-CONFIG_CDROM=y
 CONFIG_CEVT_R4K=y
 # CONFIG_CGROUPS is not set
 CONFIG_CLKBLD_I8253=y
```

Comment (guidosarducci) [reply target/linux/malta/config-5.4:37]: Sure, I will do that. Could you explain the refresh behaviour and its problem with the previous change? That wasn't clear.

**Comment by guidosarducci** [comment]
@yousong I've made the requested changes.

I should also highlight this patch was just one step in troubleshooting the larger, more serious problem of the broken malta-mips64/mipsel64 platforms (see [FS#3277](https://bugs.openwrt.org/index.php?do=details&task_id=3277)). I've tried to raise this a few times in the past (e.g. https://github.com/openwrt/packages/pull/7750#issuecomment-455738847 and https://github.com/openwrt/openwrt/commit/54124e1e1cd6a371057626c559e9cc718d24a46d#commitcomment-41097548) but received no responses.

>Also, I think we should consider re-enabling building the full set of malta configurations: 64/32-bit and be/le. They provide really useful quality coverage, and have been tremendously helpful as downloadable images available for spot testing packages or OS problems.
>
>As of today malta 64be bootloops for me , malta 64le doesn't build, and malta 32be exhausts ZONE_DMA memory during boot (with a WARN) but still runs. I still see regular commits to these platforms but it seems little manual testing is being done, and in the absence of regular buildbot builds the accumulated breakage over time will make these useless.
>
>IIRC most of the malta configs were disabled ~2 years ago, mainly out of infrastructure concerns (e.g. around snapshot storage). I know there have been a lot of improvements since then, so perhaps the original decision can be revisited.
>
Would you be willing to help reinstate the malta builds?

**Comment by yousong** [comment]
My impression is that the project is in constant need for more infra resources and the reason that malta mips64 build was disabled because its usage at that time was too marginal (only octeon) for the use of computing and storage resources (cost and gain).  Malta mips32/be build on the other hand,  if I remembered it right, was reinstated because it can be used to debug ar71xx binaries.

Remove "source-only" and restart buildbot master should suffice for including them in the ci build.  More coverage is good, but please be sure to be more specific on the "tremendously helpful" part when submitting a patch making this change.

**Comment by hauke** [comment]
removing CDROM support looks fine for me. I do not think anyone uses this. 
We should probably drop 4.19 support soon, but this is unrelated.

**Review by hauke — APPROVED** [review]

**Comment by guidosarducci** [comment]
> My impression is that the project is in constant need for more infra resources and the reason that malta mips64 build was disabled was because its usefulness at that time was too marginal (only octeon) for the use of computing and storage resources (cost and gain). Malta mips32/be build on the other hand, if I remembered it right, was reinstated because it can be used to debug ar71xx binaries.
> 
Correct, there are also targets that use the same mipsel_24kc arch as the mips32/le target, so there are valid arguments for enabling that as well, but overall I understand the need to manage resources effectively.

Regardless of automatic builds though, my main concern is with the target being available when needed, which isn't the case for malta/mips64. I've looked through the logs as you have, and noticed little documented testing taking place after malta-impacting changes. The last definitive testing I see was from a year ago: f342ffd300 ("treewide: kernel: bump some targets to 4.19").

If we could simply encourage or set standards for committers to perform and document testing, it would go a long way to keeping "source-only" target like malta "healthy". For example, the PR template for the openwrt repo says nothing about such standards, whereas the packages repo template outlines the basics.

> Remove "source-only" and restart buildbot master should suffice for including them in the ci build. More coverage is good, but please be sure to be more specific on the "tremendously helpful" part when submitting a patch making this change.
> 
Thanks, I'll look into this once the malta/mips64 issues are resolved.

> removing CDROM support looks fine for me. I do not think anyone uses this.
> We should probably drop 4.19 support soon, but this is unrelated.
>
Thanks for the ACK. Regarding 4.19, I think it best to hold off removal since malta/mips64 currently works only on this kernel.

**Review by yousong — APPROVED** [review]
I will test and apply this.

**Comment by yousong** [comment]
Applied.  Thanks

**Pull request closed without merging**
